tests/garage/tf/models/test_new_model_pickle.py

```diff
@@ -0,0 +1,63 @@
+import pickle
+import unittest
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.models.gaussian_mlp_model import GaussianMLPModel
+
+
+class TestNewModelPickling(unittest.TestCase):
+    def test_model_pickle(self):
+        sess = tf.Session()
+        with sess:
+            input_var = tf.placeholder(tf.float32, shape=(None, 5))
+            model = GaussianMLPModel(output_dim=2)
+            model.build(input_var)
+            data = np.random.random((3, 5))
+            results = sess.run(model.outputs[:2], feed_dict={input_var: data})
+            model_data = pickle.dumps(model)
+
+        sess.close()
+        tf.reset_default_graph()
+
+        new_sess = tf.Session()
+        with new_sess as sess:
+            input_var = tf.placeholder(tf.float32, shape=(None, 5))
+            model_pickled = pickle.loads(model_data)
+            model_pickled.build(input_var)
+
+            results2 = sess.run(
+                model_pickled.outputs[:2], feed_dict={input_var: data})
+
+            # result[0] is sample(), which is supposed to be different
+            assert np.array_equal(results[1:], results2[1:])
```

Comment:
Why would the sample be different if the global seed is the same?

Reply:
Since it is not inheriting from `TfGraphTestCase` and I forgot to call `ext.set_seed()`.